pkg/apis/core/register.go

```diff
@@ -96,6 +96,12 @@ func addKnownTypes(scheme *runtime.Scheme) error {
 		&RangeAllocation{},
 		&ConfigMap{},
 		&ConfigMapList{},
+		&ResourceClass{},
+		&ResourceClassList{},
+		&ResourceClaim{},
+		&ResourceClaimList{},
+		&PodScheduling{},
+		&PodSchedulingList{},
```

Comment:
That's what I had initially, but the result was a mixture of some types in core/v1 (ResourceClaimSpec) because they were needed there and other types in a separate apigroup (ResourceClaim). Tim then asked to put everything into core/v1.

Even though these fields and types will appear in core/v1, they will be documented and treated as alpha (not available when feature is off). We cannot add them only in core/v1alpha1 because then they wouldn't survive roundtrips through v1. This is how beta or GA APIs have been extended in the past.

I see that I missed the "alpha" comment in a few places. I'll fix that.

Reply:
> Even though these fields and types will appear in core/v1, they will be documented and treated as alpha (not available when feature is off). We cannot add them only in core/v1alpha1 because then they wouldn't survive roundtrips through v1. This is how beta or GA APIs have been extended in the past.

Adding specific feature-gated *fields* to GA types at different stability levels is a thing we do, but I'm not aware of adding new top-level types to a GA version and characterizing them as alpha. @thockin, is this really what you were asking for?

Also, I thought we didn't want to add new top-level types to the legacy core API group in general.

Reply:
FWIW, [Tim wrote](https://github.com/kubernetes/enhancements/pull/3064#discussion_r895081252) during KEP review:

> IIUC, it's exceptionally weird to have half the types in one group and half in another. At that point I'd just argue they should go in core.

That could mean that I need to bring back core/v1alpha1 (which no longer exists) and then promote these types through different stages until they finally land in core/v1. I can do that if that is preferred.

Reply:
However, we then have the problem again that documentation for core/v1 shows `ResourceClaimSpec` but not `ResourceClaim`, which might be confusing. it might be simpler and thus less error-prone to just put them into core/v1 right away - just my two cents.

Reply:
I think we could still argue about changes while it is marked as alpha, but as soon as we reach beta it would become very hard to make changes. core/v1alpha1 with just the three types would be cleaner, with promotion to core/v1beta1 etc. Is that okay?

Reply:
The validation/strategy/etc pretty much LGTM, but this question is very important and it doesn't have a conclusion.

@liggitt @thockin

Reply:
FWIW, I am working on moving the three types into "core.k8s.io" v1alpha1.

Reply:
I'm not sure yet whether I can use "core.k8s.io" as group name or have to use "" as for the core v1.

Reply:
The trend is to use a separate API group after all - see also https://kubernetes.slack.com/archives/C0EG7JC6T/p1667495339738469